packages/grid/_modules_/grid/hooks/features/infiniteLoader/useGridInfiniteLoader.ts

```diff
@@ -0,0 +1,35 @@
+import * as React from 'react';
+import { optionsSelector } from '../../utils/optionsSelector';
+import { GridApiRef } from '../../../models/api/gridApiRef';
+import { useGridSelector } from '../core/useGridSelector';
+import { useNativeEventListener } from '../../root/useNativeEventListener';
+import { GRID_SCROLL } from '../../../constants/eventsConstants';
+import { gridContainerSizesSelector } from '../../utils/sizesSelector';
+
+export const useGridInfiniteLoader = (
+  windowRef: React.MutableRefObject<HTMLDivElement | null>,
+  apiRef: GridApiRef,
+): void => {
+  const options = useGridSelector(apiRef, optionsSelector);
+  const containerSizes = useGridSelector(apiRef, gridContainerSizesSelector);
+  const isInScrollBottomArea = React.useRef<boolean>(false);
+
+  const handleGridScroll = React.useCallback(
+    (event) => {
+      const scrollPositionBottom =
+        event.target.scrollTop + containerSizes?.windowSizes.height + options.scrollBottomThreshold;
+
+      if (containerSizes && scrollPositionBottom >= containerSizes.dataContainerSizes.height) {
+        if (!isInScrollBottomArea.current && options.onScrollBottom) {
+          isInScrollBottomArea.current = true;
+          options.onScrollBottom();
+        }
+      } else {
```

Comment:
if you revert your first then we don't need an if in another if

Reply:
Unfortunately, this will not work. The reason why I need the `isInScrollBottomArea` flag is to make sure that the event only fires once, not every time you scroll inside the `scrollEndThresholdHeight` area.

Reply:
I think that what @dtassone meant to say is that we could potentially flatten the branching logic.

Reply:
Updated